### PR TITLE
Throw a descriptive error on invalid config file

### PR DESCRIPTION
**Pull request by smoliji**
Error: ENOENT: no such file or directory, open '.env.json'
-->
Error: Missing or invalid config file `.env.json` at path/to/project/.env.json

for both the Default config path as well as the User config path

### CODE REVIEW

File: src/lib/storage.ts [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+import * as path from 'path';
 import { ConfigLoaderOptions } from './loader';
 const fromPairs = (pairs: Array<[keyof any, any]>) => Object.assign({}, ...Array.from(pairs, ([k, v]) => ({ [k]: v })));
 const { keys } = Object;
 
+const loadFile = (filePath?: string) => {
+    try {
+        return filePath ? JSON.parse(readFileSync(filePath, 'utf-8')) : {};
+    } catch (error) {
+        throw new Error(`Missing or invalid config file \`${filePath}\` at ${path.join(path.resolve('.'), filePath!)}`);
```
Comment (grissius) [inline src/lib/storage.ts]:
```suggestion
        throw new Error(`Missing or invalid config file \`${filePath}\` at ${path.join(path.resolve('.'), filePath!)}. Update the path via \`defaultConfig\` option, or unset it to diable it.`);
```

File: src/lib/storage.ts [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+    }
+};
+
 export const createConfigStorage = (
     opts: Pick<ConfigLoaderOptions, 'defaultConfigPath' | 'userConfigPath' | 'envMode'>
 ): Record<any, any> => {
-    const defaultConfig = opts.defaultConfigPath ? JSON.parse(readFileSync(opts.defaultConfigPath, 'utf-8')) : {};
-    const userConfig = opts.userConfigPath ? JSON.parse(readFileSync(opts.userConfigPath, 'utf-8')) : {};
+    const defaultConfig = loadFile(opts.defaultConfigPath);
+    const userConfig = loadFile(opts.userConfigPath);
```
Comment (grissius) [inline src/lib/storage.ts]: :+1:

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [0.2.0] - 2019-04-30
 ### Added
 - Custom loaders
+- Descriptive errors on invalid config files
```
Comment (grissius) [inline CHANGELOG.md]: This should be in unreleased section, not in a published release 0.2.0 :scream:

File: src/lib/storage.ts (hunk 1 above)
Comment (smoliji) [reply src/lib/storage.ts]: `...or fix json formatting`

But we dont know if its the defaultConfig or userConfigPath. Logic is the same for both

Comment (grissius) [reply src/lib/storage.ts]: :roll_eyes:

**Pull request merged**